lib/LocalEnsemble.js

```diff
@@ -119,71 +124,86 @@ function LocalEnsemble(ensembleOrModels, connection, maxModels) {
   this.importance = [];
   this.modelsReadyCount = 0;
   this.opKinds = ENSEMBLE_OPERATING_KINDS;
-
+  this.maxParallelDownloads = maxParallelDownloads;
+  this.maxModels = maxModels;
+  this.downloadSlots = maxParallelDownloads;
+  this.batchOffset = 0;
+  this.ensembleReady = false;
 
   fillEnsembleInfo = function (error, resource) {
     /**
      * Fills the modelIds array from the ensemble resource info
      */
-    var index, len, index2, len2, categories;
+    var index, len, index2, len2, categories, batches;
 
-    if (error) {
-      throw new Error('Cannot create the Ensemble instance. Could not' +
-                      ' retrieve the resource: ' + error);
-    }
-    if (typeof resource.object !== 'undefined') {
-      resource = resource.object;
-    }
-    self.modelIds = resource.models;
-    self.distributions = resource.distributions || []
-    self.importance = resource.importance || []
+    if (!self.ensembleReady) {
+      if (error) {
+        throw new Error('Cannot create the Ensemble instance. Could not' +
+                        ' retrieve the resource: ' + error);
+      }
+      if (typeof resource.object !== 'undefined') {
+        resource = resource.object;
+      }
+      self.modelIds = resource.models;
+      self.distributions = resource.distributions || []
+      self.importance = resource.importance || []
 
-    if (resource.type === BOOSTING) {
-      self.boosting = resource.boosting;
-    }
-    // new ensembles have the fields structure
-    if (typeof resource.ensemble !== 'undefined'){
-      self.fields = resource.ensemble.fields;
-      self.objectiveField = resource.objective_field;
-      self.isRegression = self.fields[self.objectiveField].optype == 'numeric';
-    }
-    if (self.boosting) {
-      if (self.isRegression) {
-        self.boostingOffsets = (resource['initial_offset'] ||
-          REGRESSION_BOOSTING_OFFSET_DFT);
-      } else {
-        self.boostingOffsets = toMap(resource['initial_offsets'] ||
-          CLASSIFICATION_BOOSTING_OFFSET_DFT);
+      if (self.modelsSplits.length == 0) {
+        batches = Math.ceil((self.modelIds.length + 0.0) / self.maxModels);
```

Comment:
Is this `+ 0.0` here due to some JS quirk? what is it for?

Reply:
No problem, both seem correct.